scripts/2.TCGA-MLexample.py

```diff
@@ -105,69 +91,69 @@ def fs_mad(x, y):
     scores = mad(x) 
     return scores, np.array([np.NaN]*len(scores))
 
-# select the top features with the highest MAD
-feature_select = SelectKBest(fs_mad, k=n_feature_kept)
-
 
 # ## Define pipeline and Cross validation model fitting
 
-# In[12]:
-
-# Include loss='log' in param_grid doesn't work with pipeline somehow
-clf = SGDClassifier(random_state=0, class_weight='balanced',
-                    loss=param_fixed['loss'], penalty=param_fixed['penalty'])
+# In[11]:
 
-# joblib is used to cross-validate in parallel by setting `n_jobs=-1` in GridSearchCV
-# Supress joblib warning. See https://github.com/scikit-learn/scikit-learn/issues/6370
-warnings.filterwarnings('ignore', message='Changing the shape of non-C contiguous array')
-clf_grid = GridSearchCV(estimator=clf, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')
-pipeline = make_pipeline(
-    feature_select,  # Feature selection
-    StandardScaler(),  # Feature scaling
-    clf_grid)
+# Parameter Sweep for Hyperparameters
+param_grid = {
+    'select__k': [2000],
+    'classify__loss': ['log'],
+    'classify__penalty': ['elasticnet'],
+    'classify__alpha': [10 ** x for x in range(-3, 1)],
+    'classify__l1_ratio': [0, 0.2, 0.8, 1],
```

Comment:
One side effect of fixing the pipeline is that things are much slower as transformations are now separately run on every CV fold. On the sklearn issue we discussed caching because lot's of work is repeated ver batim, but caching is not available yet.